src/util/Util.js

```diff
@@ -236,18 +236,20 @@ class Util {
    * @param {number} [guildsPerShard=1000] Number of guilds per shard
    * @returns {Promise<number>} The recommended number of shards
    */
-  static fetchRecommendedShards(token, guildsPerShard = 1000) {
+  static async fetchRecommendedShards(token, guildsPerShard = 1000) {
     if (!token) throw new DiscordError('TOKEN_MISSING');
-    return fetch(`${DefaultOptions.http.api}/v${DefaultOptions.http.version}${Endpoints.botGateway}`, {
+    const response = await fetch(`${DefaultOptions.http.api}/v${DefaultOptions.http.version}${Endpoints.botGateway}`, {
       method: 'GET',
       headers: { Authorization: `Bot ${token.replace(/^Bot\s*/i, '')}` },
-    })
-      .then(res => {
-        if (res.ok) return res.json();
-        if (res.status === 401) throw new DiscordError('TOKEN_INVALID');
-        throw res;
-      })
-      .then(data => data.shards * (1000 / guildsPerShard));
+    });
+    if (response.ok) {
+      const { shards } = await response.json();
+      return shards * (1000 / guildsPerShard);
+    } else if (response.status === 401) {
+      throw new DiscordError('TOKEN_INVALID');
+    } else {
+      throw response;
+    }
```

Comment:
No useless else-return:
```suggestion
    }

    throw response.status === 401 ? new DiscordError('TOKEN_INVALID') : response;
```